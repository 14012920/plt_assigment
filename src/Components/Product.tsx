import React from "react";
import { Button, Text, withTheme, Card } from "react-native-paper";
import { Product } from "../Redux/ActionsType/ProductsActionsType";

const ProductCard = ({
  item,
  onPressAdd,
}: {
  item: Product;
  theme: any;
  onPressAdd: any;
}) => {
  return (
    <Card style={{ marginHorizontal: 10 }}>
      <Card.Content style={{ borderRadius: 0 }}>
        <Card.Cover
          source={{ uri: item?.img }}
          style={{ resizeMode: "contain" }}
        />
        <Text variant="titleLarge">{item?.name}</Text>
        <Text variant="headlineSmall">{`Price:${item?.price}`}</Text>
        <Text variant="bodyMedium">{`Color:${item?.colour}`}</Text>
      </Card.Content>
      <Card.Actions>
        <Button onPress={onPressAdd} accessibilityHint="add-cart-button">
          ADD+
        </Button>
      </Card.Actions>
    </Card>
  );
};
export default withTheme(ProductCard);
