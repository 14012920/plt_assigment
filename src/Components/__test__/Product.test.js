import React from "react";
import "@testing-library/jest-dom";
import { screen, render, fireEvent } from "@testing-library/react-native";
import { useSelector, useDispatch } from "react-redux";
import renderer from "react-test-renderer";
import Product from "../Product";

jest.mock("react-native-safe-area-context", () => {
  const inset = { top: 0, right: 0, bottom: 0, left: 0 };

  return {
    SafeAreaProvider: jest.fn().mockImplementation(({ children }) => children),
    SafeAreaConsumer: jest
      .fn()
      .mockImplementation(({ children }) => children(inset)),
    useSafeAreaInsets: jest.fn().mockImplementation(() => inset),
  };
});
const mockDispatch = jest.fn();
jest.mock("react-redux", () => ({
  useSelector: jest.fn(),
  useDispatch: () => mockDispatch,
}));
const props = {
  item: {
    id: 1,
    colour: "Black",
    name: "Black Sheet Strappy Textured Glitter Bodycon Dress",
    price: 10,
    img: "http://cdn-img.prettylittlething.com/9/0/a/a/90aa90903a135ee59594f47c7685aa7ef3046e44_cly8063_1.jpg?imwidth=1024",
    qty: 1,
  },
  onPressAdd: jest.fn(),
};
describe("render Product Screen", () => {
  beforeEach(() => {});

  test("snapshot of product screen", () => {
    const tree = renderer.create(<Product {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("render Header for product Screen", () => {
    const { getByText } = render(<Product {...props} />);
    const button = getByText("ADD+");
    fireEvent.press(button);
    expect(props.onPressAdd).toHaveBeenCalled();
  });
});
