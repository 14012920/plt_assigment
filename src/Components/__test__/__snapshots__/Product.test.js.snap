// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Header Screen render Header Screen 1`] = `
<View
  collapsable={false}
  style={
    {
      "alignSelf": undefined,
      "bottom": undefined,
      "left": undefined,
      "position": undefined,
      "right": undefined,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 3,
      "top": undefined,
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(247, 243, 249, 1)",
          "borderRadius": 12,
          "elevation": 1,
          "marginHorizontal": 10,
        }
      }
    >
      <View
        accessibilityState={
          {
            "disabled": true,
          }
        }
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "flexShrink": 1,
          }
        }
        testID="card"
      >
        <View
          style={
            [
              {
                "paddingHorizontal": 16,
              },
              {
                "paddingTop": 16,
              },
              {
                "borderRadius": 0,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "backgroundColor": "#eeeeee",
                  "height": 195,
                  "overflow": "hidden",
                },
                {
                  "borderRadius": 12,
                },
                {
                  "resizeMode": "contain",
                },
              ]
            }
          >
            <Image
              accessibilityIgnoresInvertColors={true}
              source={
                {
                  "uri": "http://cdn-img.prettylittlething.com/9/0/a/a/90aa90903a135ee59594f47c7685aa7ef3046e44_cly8063_1.jpg?imwidth=1024",
                }
              }
              style={
                [
                  {
                    "flex": 1,
                    "height": undefined,
                    "justifyContent": "flex-end",
                    "padding": 16,
                    "width": undefined,
                  },
                  {
                    "borderRadius": 12,
                  },
                ]
              }
            />
          </View>
          <Text
            style={
              [
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontSize": 22,
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 28,
                },
                {
                  "textAlign": "left",
                },
                {
                  "writingDirection": "ltr",
                },
                undefined,
              ]
            }
          >
            Black Sheet Strappy Textured Glitter Bodycon Dress
          </Text>
          <Text
            style={
              [
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontSize": 24,
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 32,
                },
                {
                  "textAlign": "left",
                },
                {
                  "writingDirection": "ltr",
                },
                undefined,
              ]
            }
          >
            Price:10
          </Text>
          <Text
            style={
              [
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "400",
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                },
                {
                  "textAlign": "left",
                },
                {
                  "writingDirection": "ltr",
                },
                undefined,
              ]
            }
          >
            Color:Black
          </Text>
        </View>
        <View
          index={1}
          siblings={
            [
              "Card.Content",
              "Card.Actions",
            ]
          }
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "padding": 8,
              },
              undefined,
              {
                "justifyContent": "flex-end",
              },
            ]
          }
          total={2}
        >
          <View
            collapsable={false}
            style={
              {
                "alignSelf": undefined,
                "bottom": undefined,
                "left": undefined,
                "position": undefined,
                "right": undefined,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
                "top": undefined,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "backgroundColor": "transparent",
                    "borderColor": "rgba(121, 116, 126, 1)",
                    "borderRadius": 20,
                    "borderStyle": "solid",
                    "borderWidth": 1,
                    "marginLeft": 8,
                    "minWidth": 64,
                  }
                }
              >
                <View
                  accessibilityHint="add-cart-button"
                  accessibilityRole="button"
                  accessibilityState={
                    {
                      "disabled": false,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "overflow": "hidden",
                      },
                      false,
                      {
                        "borderRadius": 20,
                      },
                    ]
                  }
                  testID="button"
                >
                  <View
                    style={
                      [
                        {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "justifyContent": "center",
                        },
                        undefined,
                      ]
                    }
                  >
                    <Text
                      numberOfLines={1}
                      selectable={false}
                      style={
                        [
                          {
                            "color": "rgba(28, 27, 31, 1)",
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          {
                            "textAlign": "left",
                          },
                          {
                            "writingDirection": "ltr",
                          },
                          [
                            {
                              "marginHorizontal": 16,
                              "marginVertical": 9,
                              "textAlign": "center",
                            },
                            false,
                            {
                              "marginHorizontal": 24,
                              "marginVertical": 10,
                            },
                            false,
                            false,
                            {
                              "color": "rgba(103, 80, 164, 1)",
                              "fontFamily": "System",
                              "fontSize": 14,
                              "fontWeight": "500",
                              "letterSpacing": 0.1,
                              "lineHeight": 20,
                            },
                            undefined,
                          ],
                        ]
                      }
                    >
                      ADD+
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`render Product Screen snapshot of product screen 1`] = `
<View
  collapsable={false}
  style={
    {
      "alignSelf": undefined,
      "bottom": undefined,
      "left": undefined,
      "position": undefined,
      "right": undefined,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 3,
      "top": undefined,
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(247, 243, 249, 1)",
          "borderRadius": 12,
          "elevation": 1,
          "marginHorizontal": 10,
        }
      }
    >
      <View
        accessibilityState={
          {
            "disabled": true,
          }
        }
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "flexShrink": 1,
          }
        }
        testID="card"
      >
        <View
          style={
            [
              {
                "paddingHorizontal": 16,
              },
              {
                "paddingTop": 16,
              },
              {
                "borderRadius": 0,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "backgroundColor": "#eeeeee",
                  "height": 195,
                  "overflow": "hidden",
                },
                {
                  "borderRadius": 12,
                },
                {
                  "resizeMode": "contain",
                },
              ]
            }
          >
            <Image
              accessibilityIgnoresInvertColors={true}
              source={
                {
                  "uri": "http://cdn-img.prettylittlething.com/9/0/a/a/90aa90903a135ee59594f47c7685aa7ef3046e44_cly8063_1.jpg?imwidth=1024",
                }
              }
              style={
                [
                  {
                    "flex": 1,
                    "height": undefined,
                    "justifyContent": "flex-end",
                    "padding": 16,
                    "width": undefined,
                  },
                  {
                    "borderRadius": 12,
                  },
                ]
              }
            />
          </View>
          <Text
            style={
              [
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontSize": 22,
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 28,
                },
                {
                  "textAlign": "left",
                },
                {
                  "writingDirection": "ltr",
                },
                undefined,
              ]
            }
          >
            Black Sheet Strappy Textured Glitter Bodycon Dress
          </Text>
          <Text
            style={
              [
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontSize": 24,
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 32,
                },
                {
                  "textAlign": "left",
                },
                {
                  "writingDirection": "ltr",
                },
                undefined,
              ]
            }
          >
            Price:10
          </Text>
          <Text
            style={
              [
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "400",
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                },
                {
                  "textAlign": "left",
                },
                {
                  "writingDirection": "ltr",
                },
                undefined,
              ]
            }
          >
            Color:Black
          </Text>
        </View>
        <View
          index={1}
          siblings={
            [
              "Card.Content",
              "Card.Actions",
            ]
          }
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "padding": 8,
              },
              undefined,
              {
                "justifyContent": "flex-end",
              },
            ]
          }
          total={2}
        >
          <View
            collapsable={false}
            style={
              {
                "alignSelf": undefined,
                "bottom": undefined,
                "left": undefined,
                "position": undefined,
                "right": undefined,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
                "top": undefined,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "backgroundColor": "transparent",
                    "borderColor": "rgba(121, 116, 126, 1)",
                    "borderRadius": 20,
                    "borderStyle": "solid",
                    "borderWidth": 1,
                    "marginLeft": 8,
                    "minWidth": 64,
                  }
                }
              >
                <View
                  accessibilityHint="add-cart-button"
                  accessibilityRole="button"
                  accessibilityState={
                    {
                      "disabled": false,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "overflow": "hidden",
                      },
                      false,
                      {
                        "borderRadius": 20,
                      },
                    ]
                  }
                  testID="button"
                >
                  <View
                    style={
                      [
                        {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "justifyContent": "center",
                        },
                        undefined,
                      ]
                    }
                  >
                    <Text
                      numberOfLines={1}
                      selectable={false}
                      style={
                        [
                          {
                            "color": "rgba(28, 27, 31, 1)",
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          {
                            "textAlign": "left",
                          },
                          {
                            "writingDirection": "ltr",
                          },
                          [
                            {
                              "marginHorizontal": 16,
                              "marginVertical": 9,
                              "textAlign": "center",
                            },
                            false,
                            {
                              "marginHorizontal": 24,
                              "marginVertical": 10,
                            },
                            false,
                            false,
                            {
                              "color": "rgba(103, 80, 164, 1)",
                              "fontFamily": "System",
                              "fontSize": 14,
                              "fontWeight": "500",
                              "letterSpacing": 0.1,
                              "lineHeight": 20,
                            },
                            undefined,
                          ],
                        ]
                      }
                    >
                      ADD+
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
